### ISSUE TITLE
faster geo calculations (feature request)

### ISSUE THREAD

**Issue by kkdd** [arangodb/arangodb#4368]
Hello,
This is something like feature request (or an idea), derived from reading https://github.com/arangodb/arangodb/blob/devel/arangod/MMFiles/mmfiles-geo-index.cpp.

For faster geo calculations, don't use (or calculate) great-circle distance, \d, but use (the half of) direct distance, \tilde{d}, which also satisfies triangular inequality, 
and don't use longitude or latitude, but use the half of so-called "direct"-longitude and latitude as described by the following relations:

```latex
\tilde{d} &\triangleq \sin \frac{d}{2} \\
\tilde{\phi} &\triangleq \sin \frac{\phi}{2} \\
\tilde{\lambda} &\triangleq \sin \frac{\lambda}{2} \\
\tilde{d}^2 &= \sin^2 \frac{\phi_1 - \phi_2}{2} + \sin^2 \frac{\lambda_1 - \lambda_2}{2} \cos \phi_1 \cos \phi_2 \\
&= \left( \tilde{\phi}_1 \cos \frac{\phi_2}{2} - \tilde{\phi}_2 \cos \frac{\phi_1}{2} \right)^2 + \left( \tilde{\lambda}_1 \cos \frac{\lambda_2}{2} - \tilde{\lambda}_2 \cos \frac{\lambda_1}{2} \right)^2 \cos \phi_1 \cos \phi_2 \\
\cos \phi &= 1 - 2 {\tilde{\phi}}^2 \\
\cos \frac{\phi}{2} &= \sqrt{1 - {\tilde{\phi}}^2} \\
\cos \frac{\lambda}{2} &= \sqrt{1 - {\tilde{\lambda}}^2} \\
```

To avoid degradation of precision at the neigbor of \lambda = \pm \pi, use \sin \frac{\lambda}{4} rather than \sin \frac{\lambda}{2}, or use \tan \frac{\phi}{4} and \tan \frac{\lambda}{4}, instead.

```latex
\chi &\triangleq \tan \frac{\theta}{2} \\
\cos \theta &= \frac{1 - \chi^2 }{1 + \chi^2} \\ 
\sin \theta &= \frac{2 \chi}{1 + \chi^2} \\
```

In summary, don't use an angle itself but use its substitutions for faster geo calculations.

**Comment by dothebart** [comment]
Meanwhile the geo-Index is using the S2 library. 

ArangoDB 3.7 is Available for download:
https://www.arangodb.com/2020/08/arangodb-3-7-a-big-step-forward-for-multi-model/

Closing as outdated.
